File: src/main/resources/static/js/service/announcmentService.js
//服务层
app.service('announcmentService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('../announcment/findAll');		
	}
	//分页 
	this.findPage=function(page,rows){
		return $http.get('../announcment/findPage?page='+page+'&rows='+rows);
	}
	//查询实体
	this.findOne=function(id){
		return $http.get('../announcment/findOne?id='+id);
	}
	//增加 
	this.add=function(entity){
		return  $http.post('../announcment/add',entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post('../announcment/update',entity );
	}
	//删除
	this.dele=function(ids){
		return $http.get('../announcment/delete?ids='+ids);
	}
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post('../announcment/search?page='+page+"&rows="+rows, searchEntity);
	}    	
});

File: src/main/java/com/ty/lab/service/AnnouncmentService.java
package com.ty.lab.service;

import com.ty.lab.entity.PageResult;
import com.ty.lab.pojo.TbAnnouncment;

import java.util.List;

/**
 * 服务层接口
 *
 * @author Administrator
 */
public interface AnnouncmentService {

    /**
     * 返回全部列表
     *
     * @return
     */
    public List<TbAnnouncment> findAll();

    /**
     * 返回分页列表
     *
     * @return
     */
    public PageResult findPage(int pageNum, int pageSize);

    /**
     * 增加
     */
    public void add(TbAnnouncment announcment);

    /**
     * 修改
     */
    public void update(TbAnnouncment announcment);

    /**
     * 根据ID获取实体
     *
     * @param id
     * @return
     */
    public TbAnnouncment findOne(Long id);

    /**
     * 批量删除
     *
     * @param ids
     */
    public void delete(Long[] ids);

    /**
     * 分页
     *
     * @param pageNum  当前页 码
     * @param pageSize 每页记录数
     * @return
     */
    public PageResult findPage(TbAnnouncment announcment, int pageNum, int pageSize);

}
